### PR TITLE
SR 1.5

### PR DESCRIPTION
**Pull request by krogla**
StakingRouter 1.5 Upgrade, deployed on mainnet

### CODE REVIEW

File: contracts/0.8.9/sanity_checks/OracleReportSanityChecker.sol [hunk 1]
```diff
@@ -309,25 +340,56 @@ contract OracleReportSanityChecker is AccessControlEnumerable {
...
         external
-        onlyRole(MAX_ACCOUNTING_EXTRA_DATA_LIST_ITEMS_COUNT_ROLE)
+        onlyRole(SECOND_OPINION_MANAGER_ROLE)
     {
         LimitsList memory limitsList = _limits.unpack();
-        limitsList.maxAccountingExtraDataListItemsCount = _maxAccountingExtraDataListItemsCount;
+        limitsList.clBalanceOraclesErrorUpperBPLimit = _clBalanceOraclesErrorUpperBPLimit;
         _updateLimits(limitsList);
+        if (_secondOpinionOracle != secondOpinionOracle) {
+            secondOpinionOracle = ISecondOpinionOracle(_secondOpinionOracle);
```
Comment (louisbrown0212) [inline contracts/0.8.9/sanity_checks/OracleReportSanityChecker.sol:378]: nit: Type conversion is not necessary

**Review by tamtamchik — APPROVED** [review]
🎉

File: contracts/0.8.9/sanity_checks/OracleReportSanityChecker.sol [hunk 2]
```diff
@@ -559,16 +629,101 @@ contract OracleReportSanityChecker is AccessControlEnumerable {
...
+        }
+        return sum;
+    }
+
+    function _exitedValidatorsAtTimestamp(uint256 _timestamp) internal view returns (uint256) {
+        for (int256 index = int256(reportData.length) - 1; index >= 0; index--) {
+            if (reportData[uint256(index)].timestamp <= SafeCast.toUint64(_timestamp)) {
+                return reportData[uint256(index)].totalExitedValidators;
+            }
+        }
```
Comment (louisbrown0212) [inline contracts/0.8.9/sanity_checks/OracleReportSanityChecker.sol:657]:
```suggestion
        for (uint256 index = reportData.length - 1; ; index--) {
            if (reportData[index].timestamp <= SafeCast.toUint64(_timestamp)) {
                return reportData[index].totalExitedValidators;
            }
            if (index == 0) {
                break;
            }
        }
```

Comment (krogla) [reply contracts/0.8.9/sanity_checks/OracleReportSanityChecker.sol:657]: I agree with the comments, but we have this exact code already deployed in mainnet, and we need to maintain consistency between the code in the master branch and the verified deployed bytecode. So suggest moving these changes to a separate PR in develop branch

**Review by TheDZhon — APPROVED** [review]
I confirm zero contract changes against the deployed and audited version :saluting_face: 

The reference commit hash: [`1ffbb7e49e112fcac678f59bf63ba57a7e522874`](https://github.com/lidofinance/core/tree/1ffbb7e49e112fcac678f59bf63ba57a7e522874)

Audit reports:
- [MixBytes](https://github.com/lidofinance/audits/blob/main/MixBytes%20Lido%20CSM%20Security%20Audit%20Report%2010-24.pdf)
- [Ackee Blockchain](https://github.com/lidofinance/audits/blob/main/Ackee%20Blockchain%20Lido%20Staking%20Router%20v2%20Report%2010-24.pdf)

**Pull request merged**